### PR TITLE
Added granularity to the report charts

### PR DESCRIPTION
**Pull request by brumik**
### Problems:
* Chart x axis labels are not too helpful when setting to year (showing 01/01 for all ticks)
  * possible solutions: display full date always or change the date depending on the granularity (hard mode)
* Quick date range from API does not change with the granularity which is then weird
* Granularity could go next to the date selector instead to the category drop down. Waiting for the #555 PR to be merged, so we don't cause accidental bugs.

(Done) TODO: 
- Granularity to the left of the date selector
- Change the chart axis depending on the granularity

### CODE REVIEW

**Comment by h-kataria** [comment]
LGTM, tested in UI. @jlmitch5 i defer it to you for approval

**Comment by brumik** [comment]
This is a Draft PR. I need feedback on the description. Do not merge yet.

File: src/Utilities/useQueryParams.js [hunk 1]
```diff
@@ -43,6 +43,33 @@ export const useQueryParams = (initial) => {
...
+            return {
+              ...state,
+              ...value,
+              quick_date_range: 'last_6_months',
+            };
+          case 'yearly':
+            return {
+              ...state,
+              ...value,
+              quick_date_range: 'last_6_years',
```
Comment (jlmitch5) [inline src/Utilities/useQueryParams.js:64]: @trahman73 what would you say a good default for the yearly granular setting would be? Are options are Past 1, 3, 6 and 10 years

File: src/Utilities/useQueryParams.js [hunk 2]
```diff
@@ -58,16 +85,10 @@ export const useQueryParams = (initial) => {
...
-        if (value !== 'custom') {
-          newState = { ...newState, start_date: null, end_date: null };
-        }
-
-        newState = { ...newState, ...value };
-        return newState;
-      }
-
+      case 'SET_QUICK_DATE_RANGE':
+        return value !== 'custom'
```
Comment (jlmitch5) [inline src/Utilities/useQueryParams.js:89]: is this just a bug fix?  or is it related to the granularity?

**Review by jlmitch5 — COMMENTED** [review]
I pulled this down and it looks great!

My only question is if we should do a different default for yearly granularity.  I'm curious what @trahman73 thinks.

Also I wasn't quite sure about the useQueryParams change for the quick date range setting...was that just fixing a bug?

File: src/Utilities/useQueryParams.js (hunk 1 above)
Comment (trahman73) [reply src/Utilities/useQueryParams.js:64]: I would say Past 1 as the default.

Comment (brumik) [reply src/Utilities/useQueryParams.js:64]: @trahman73 I don't think that would be wise, that is one tick which means no lines in the chart just dots :) so it is pretty useless for the user.

File: src/Utilities/useQueryParams.js (hunk 2 above)
Comment (brumik) [reply src/Utilities/useQueryParams.js:89]: I had some weird bug there and I don't know why it got fixed by removing the intermediate `let` variable. Slightly worrying that I don't know the reason but tested it and works so fine by me.

File: src/Utilities/useQueryParams.js (hunk 1 above)
Comment (trahman73) [reply src/Utilities/useQueryParams.js:64]: @brumik Ah, I didn't realize that would result in only a single tick. Then I would say 6 or even 10 if it fits comfortably in the view.

Comment (brumik) [reply src/Utilities/useQueryParams.js:64]: I set 6 as a good middle ground (I tried to set the second biggest for each granularity for consistency)

**Review by h-kataria — APPROVED** [review]

**Pull request merged**